### PR TITLE
[decomp] `loader` prelim work + some cleanup

### PR DESCRIPTION
**Pull request by ManDude**

### CODE REVIEW

File: decompiler/config/all-types.gc [hunk 1]
```diff
@@ -4482,14 +4486,14 @@
...
-    (dummy-11 () none 11)
-    (dummy-12 () none 12)
-    (dummy-13 () none 13)
-    (dummy-14 () none 14)
-    (dummy-15 () none 15)
-    (dummy-16 () none 16)
+    (update (_type_ symbol) int 9)
+    (clear-rec (_type_) int 10)
+    (spool-push (_type_ string int process-drawable float) int 11)
+    (first-using-status (_type_ string int) symbol 12)
```
Comment (ManDude) [inline decompiler/config/all-types.gc:4492]: rename this

File: goal_src/engine/level/level.gc [hunk 2]
```diff
@@ -421,7 +421,7 @@
                 )
               )
              (else
-              ;; the background is always the last object, so we're loading the background now
+              ;; so we're loading the last object now, which has different rules
```
Comment (ManDude) [inline goal_src/engine/level/level.gc:424]: typo

File: goal_src/kernel/gkernel-h.gc [hunk 3]
```diff
@@ -571,3 +571,16 @@
       `(define ,name (lambda :name ,name :behavior ,process-type ,bindings ,@body))
       )
   )
+
+(defmacro handle->name (handle)
+  "get the name of a process using a handle. #f is the result if the handle was invalid."
+
+  (with-gensyms (,proc)
```
Comment (ManDude) [inline goal_src/kernel/gkernel-h.gc:578]: wrong.

File: goal_src/engine/load/loader.gc [hunk 4]
```diff
@@ -5,3 +5,1004 @@
...
+    (set! (-> obj rec 2 name) name)
+    (set! (-> obj rec 2 parts) part)
+    (set! (-> obj rec 2 priority) priority)
+    (set! (-> obj rec 2 owner) (process->handle proc))
+    )
+   )
+  0
+  )
+
+(defun ja-play-spooled-anim ((arg0 spool-anim) (arg1 art-joint-anim) (arg2 art-joint-anim) (arg3 (function process-drawable symbol)))
```
Comment (ManDude) [inline goal_src/engine/load/loader.gc:764]: use behavior

**Review by water111 — APPROVED** [review]
Nice!

**Pull request merged**